src/classes/routes/RouterFactory.ts

```diff
@@ -0,0 +1,60 @@
+import { Router, Request, Response, NextFunction } from 'express';
+import PostImageController from './PostImageController';
+import Logger from '../Logger';
+import ConfigurationManager from '../ConfigurationManager';
+import * as multer from 'multer';
+import * as cors from 'cors';
+import { ExpressMiddleware } from '../../types';
+
+class RouterFactory {
+
+    constructor (private configurationManager: ConfigurationManager, private logger: Logger) { }
+
+    public buildImageRouter (): Router {
+        const router: Router = Router();
+        const fileInputName = this.configurationManager.getFileInputName();
+        const uploadFilePath = this.configurationManager.getUploadPath();
+
+        router.use(this.createCorsMiddleware());
+
+        router.options('*', this.createCorsMiddleware());
+
```

Comment:
Can you create a variable for CorsMiddleware? this.createCorsMiddleware() is called two times here.